src/main/java/core/basesyntax/utils/transaction/OperationFactory.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.utils.transaction;
+
+import core.basesyntax.exceptions.UnknownOperationException;
+
+public class OperationFactory {
+    public static FruitTransaction.Operation getOperationFromString(String operationString) {
+        return switch (operationString) {
+            case "b" -> FruitTransaction.Operation.BALANCE;
+            case "s" -> FruitTransaction.Operation.SUPPLY;
+            case "p" -> FruitTransaction.Operation.PURCHASE;
+            case "r" -> FruitTransaction.Operation.RETURN;
+            default -> throw new UnknownOperationException("The operation provided is invalid");
```

Comment:
Avoid using switch-cases and if-else constructions. It is recommended to use the Strategy pattern instead. Consider creating a map of strategies where the key is the operation string and the value is the corresponding `FruitTransaction.Operation`. This will allow you to look up the operation directly without the need for a switch-case.